docs/src/main/asciidoc/deploying-to-kubernetes.adoc

```diff
@@ -735,3 +736,87 @@ This assumes that a `.kube/config` is available in your user directory that poin
+Those resources can be added under `src/main/kubernetes` and can be named after the target environment (e.g. kubernetes.json, openshift.json, knative.json, or the yml equivalents). Each of these files may contain one or more Kubernetes resources.
+
+Any resource found, will be added in the generated manifests. Global modifications (e.g. labels, annotations etc) will also be applied to those resources.
+If one of the provided resources has the same name with one of the generated resources, then the generated resource will be created on top of the provided, respecting existing content when possible (e.g. existing labels, annotations, environment variables, mounts, replicas etc).
+
+The name of the resource is determined by the application name, and may be overridden by `quarkus.kubernetes.name`, `quarkus.openshift.name` and `quarkus.knative.name`.
+
+For example, in the kubernetes-quickstart application, we can add a `kuberenetes.yml` file in the `src/main/kubernetes` that looks like:
+
+[source,yaml]
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kubernetes-quickstart
+  labels:
+    app: quickstart
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: quickstart
+  template:
+    metadata:
+      labels:
+        app: quickstart
+    spec:
+      containers:
+      - name: kubernetes-quickstart
+        image: someimage:latest
+        ports:
+        - containerPort: 80
+        env:
+        - name: FOO
+          value: BAR
+---
+
+The generated `kubernetes.yml` will look like:
+
+[source,yaml]
+---
+apiVersion: "apps/v1"
+kind: "Deployment"
+metadata:
+  annotations:
+    app.quarkus.io/build-timestamp: "2020-04-10 - 12:54:37 +0000"
+  labels:
+    app: "greeting"
+  name: "kubernetes-quickstart"
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: "kubernetes-quickstart"
+      app.kubernetes.io/version: "1.0-SNAPSHOT"
+  template:
+    metadata:
+      annotations:
+        app.quarkus.io/build-timestamp: "2020-04-10 - 12:54:37 +0000"
+      labels:
+        app: "greeting"
+    spec:
+      containers:
+      - env:
+        - name: "FOO"
+          value: "BAR"
+        image: "<<yourDockerUsernmae>>/kubernetes-quickstart:1.0-SNAPSHOT"
+        imagePullPolicy: "Always"
+        name: "kubernetes-quickstart"
+        ports:
+        - containerPort: 8080
+          name: "http"
+          protocol: "TCP"
+      serviceAccount: "kubernetes-quickstart"
+---
+
+The provided, replicas, environment variables and labels where retained. However, the image and container port where modified. Also the standard annotations have been added.
+
+Note: If the resource name does not match the application name (or the overridden name) instead of reusing the resource a new one will be added. Similar is the case of the container.
```

Comment:
```suggestion
Note: If the resource name does not match the application name (or the overridden name) instead of reusing the resource a new one will be added. Same goes for the container.
```